packages/react-hooks/src/index.tsx

```diff
@@ -0,0 +1,277 @@
+  }
+  return credential
+}
+
+export const useCredentialByState = (state: CredentialState): CredentialRecord[] => {
+  const credentialState = useContext(CredentialContext)
+  const credentials = credentialState.credentials.filter((c: CredentialRecord) => c.state === state)
+  return credentials
+}
+
+// Proofs
+export const useProofs = (): { proofs: ProofRecord[]; loading: boolean } => {
+  return useContext(ProofContext)
+}
+
+export const useProofById = (id: string): ProofRecord | undefined => {
+  const { proofs } = useContext(ProofContext)
+  const proof = proofs.find((p: ProofRecord) => p.id === id)
+  if (!proof) {
+    console.warn(TAG, `Could not find Proof By ID ${id}`)
+  }
+  return proof
+}
+
+export const useProofByState = (state: ProofState): ProofRecord[] => {
+  const proofState = useContext(ProofContext)
+  const proofs = proofState.proofs.filter((p: ProofRecord) => p.state === state)
+  return proofs
+}
+
+//BasicMessages
+export const useBasicMessagesByConnectionId = (connectionId: string): BasicMessageRecord[] => {
+  const { basicMessages } = useContext(BasicMessageContext)
+  const connectionMessages = basicMessages.filter((m: BasicMessageRecord) => m.connectionId === connectionId)
+  return connectionMessages
+}
+
+interface Props {
+  agent: Agent | undefined
+  children: any
+}
+
+const AgentProvider: React.FC<Props> = ({ agent, children }) => {
+  const [agentState, setAgentState] = useState<{
+    agent: Agent | null
+    loading: boolean
+  }>({
+    agent: null,
+    loading: true,
+  })
+  const [connectionState, setConnectionState] = useState<{ connections: ConnectionRecord[] | []; loading: boolean }>({
+    connections: [],
+    loading: true,
+  })
+  const [credentialState, setCredentialState] = useState<{ credentials: CredentialRecord[] | []; loading: boolean }>({
+    credentials: [],
+    loading: true,
+  })
+  const [proofState, setProofState] = useState<{ proofs: ProofRecord[] | []; loading: boolean }>({
+    proofs: [],
+    loading: true,
+  })
+  const [basicMessageState, setBasicMessageState] = useState<{
+    basicMessages: BasicMessageRecord[] | []
+    loading: boolean
+  }>({ basicMessages: [], loading: true })
+
+  useEffect(() => {
+      setInitialState()
+  }, [agent])
+
+  const setInitialState = async () => {
+    try {
+      if(agent) {
+        const connections = await agent.connections.getAll()
+        const credentials = await agent.credentials.getAll()
+        const proofs = await agent.proofs.getAll()
+        const basicMessages = await agent.basicMessages.findAllByQuery({})
+
+        setAgentState({ agent, loading: false })
+        setConnectionState({ connections, loading: false })
+        setCredentialState({ credentials, loading: false })
+        setProofState({ proofs, loading: false })
+        setBasicMessageState({ basicMessages, loading: false })
+      }
+    } catch (e) {
+      console.error(TAG, 'ERROR IN SET_INITIAL_STATE:', e)
+    }
+  }
+
+  useEffect(() => {
```

Comment:
fine to merge now, but I think it would be nice to split into a separate file per feature (connections, credentials, proofs, etc..)